### ISSUE TITLE
Script IDE

### ISSUE THREAD

**Issue by torusrxxx** [x64dbg/x64dbg#2553]
The user should be able to create and edit a script in x64dbg and without saving them.

**Comment by mrexodia** [comment]
I think this should be out of scope of the debugger. The scripting language isn't good to begin with, so I don't see a reason to introduce a whole IDE into the scope of the debugger.

I think it could be a good idea to expose the right APIs to allow a plugin to do this though. Preferably generic so that a Python/Lua/whatever IDE could also be implemented.

**Comment by torusrxxx** [comment]
In fact I think this feature is very small in code size: just replace the table with a editable text box and done.

**Comment by mrexodia** [comment]
Yeah but then you also have to deal with the highlighting of the current
script IP, labels etc. Also the editing and loading of the script has to be
separated somehow. Perhaps with the right API it could be worthwhile, but
if it’s just for the built in language. Likely after it’s added there will
be further feature requests and generally I think it would increase the
maintenance burden even more.

The way I use the scripting it’s usually either to generate some
label/breakpoint commands or just write a quick script. In either case the
paste from clipboard feature works, but for editing files the reload button
is instant and works great. I really don’t think it’s a good language to
script complex operations, so an IDE seems overkill.

Much better would be to use Zerobrane and integrate it with the Lua plugin
(x64dbgplaytime). I did a simple POC a while back and you can use a proper
IDE to launch and debug scripts within the x64dbg context. The script API
need some work, but since my scripts are very limited the current thing
works great already.

**Comment by torusrxxx** [comment]
So when editing the script, we just use a plain text box where none of the highlighting etc are supported? Then we use a right click menu item to switch between the table (when you execute the script) and the text box (when you edit it). It is super annoying when there is an error in the script and it pauses, but there is no "assemble" command, and you have to restart the script!
